### PR TITLE
Ingest contento articles

### PR DESCRIPTION
**Pull request by mconnell**
This PR allows us to fetch article content from contento in addition to content from the blog.

* Renamed `SourceDocument` to a more explicit name `XmlSourceDocument`
* Introduced a `JsonSourceDocument`
* Added a `content_type` attribute to `Source`
* Moved some fixtures into folders to tidy things up.

### CODE REVIEW

**Comment by asmega** [comment]
:+1:

File: app/models/json_source_document.rb [hunk 1]
```diff
@@ -0,0 +1,19 @@
+require 'open-uri'
+
+class JsonSourceDocument
+  def initialize(url)
+    @document = JSON.parse(open(url).read)
```
Comment (neoeno) [inline app/models/json_source_document.rb]: Just to check — is `url` (& what it returns) trusted? Both to be non-malicious, and valid.

File: app/models/json_source_document.rb [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+  end
+
+  def items
+    @document.map do |json_item|
+      {
+        title:        json_item['label'],
+        body:         json_item['blocks'][0]['content'],
+        url:          "https://www.moneyadviceservice.org.uk#{json_item['full_path']}",
```
Comment (neoeno) [inline app/models/json_source_document.rb]: Should this URL be a constant? Or configurable?

File: app/models/json_source_document.rb (hunk 1 above)
Comment (mconnell) [inline app/models/json_source_document.rb]: The url comes from an instance of Source, which is currently a class non-accessible to the admin user, and are defined in seeds.rb. All source urls are strings representing data sources for mas content eg. The blog, contento

File: app/models/json_source_document.rb (hunk 2 above)
Comment (mconnell) [inline app/models/json_source_document.rb]: It could be a constant. It doesn't need to be configurable.

**Comment by urfolomeus** [comment]
I like the command pattern used to select the right parser based on the content type. I also think it's a sensible choice to use webmock to simulate the actual parsing in the tests. It makes it easy to see the process.

:+1:

**Comment by JiggyPete** [comment]
:+1:

**Pull request merged**